homeassistant/components/camera/__init__.py

```diff
@@ -207,6 +218,12 @@ def state_attributes(self):
 
         return attr
 
+    @callback
+    def async_update_token(self):
+        """Update the used token."""
+        self.access_tokens.append(
+            hashlib.sha256(str.encode(str(_RND.random()))).hexdigest())
```

Comment:
This is certainly an improvement, but if you're passing it through SHA256 anyway, you might as well make that the entropy 'bottleneck'.

```
        self.access_tokens.append(
             hashlib.sha256(
                 _RND.getrandbits(256).to_bytes(32, 'little')
             ).hexdigest()
        )
```

(You could also replace `_RND.getrandbits(256).to_bytes(32, 'little')` with `os.urandom(32)`)

SystemRandom().random() returns a 64 bit float, which gets you ~52 bits of entropy from the mantissa. So currently that's the bottleneck.